CI fails with min versions: Validation error for field 'import_name'

CI fails with min versions: https://github.com/huggingface/trl/actions/runs/24296581894/job/70942509620
> Validation error for field 'import_name':
> TypeError: Unsupported type for field 'import_name': str | None

```python
 ERROR tests/distributed/test_distributed.py - huggingface_hub.errors.StrictDataclassFieldValidationError: Validation error for field 'import_name':
    TypeError: Unsupported type for field 'import_name': str | None
ERROR tests/distributed/test_distributed.py - huggingface_hub.errors.StrictDataclassFieldValidationError: Validation error for field 'import_name':
    TypeError: Unsupported type for field 'import_name': str | None
ERROR tests/distributed/test_distributed.py - huggingface_hub.errors.StrictDataclassFieldValidationError: Validation error for field 'import_name':
    TypeError: Unsupported type for field 'import_name': str | None
ERROR tests/distributed/test_distributed.py - huggingface_hub.errors.StrictDataclassFieldValidationError: Validation error for field 'import_name':
    TypeError: Unsupported type for field 'import_name': str | None
ERROR tests/test_activation_offloading.py - huggingface_hub.errors.StrictDataclassFieldValidationError: Validation error for field 'import_name':
    TypeError: Unsupported type for field 'import_name': str | None
ERROR tests/test_activation_offloading.py - huggingface_hub.errors.StrictDataclassFieldValidationError: Validation error for field 'import_name':
    TypeError: Unsupported type for field 'import_name': str | None
ERROR tests/test_activation_offloading.py - huggingface_hub.errors.StrictDataclassFieldValidationError: Validation error for field 'import_name':
    TypeError: Unsupported type for field 'import_name': str | None
ERROR tests/test_activation_offloading.py - huggingface_hub.errors.StrictDataclassFieldValidationError: Validation error for field 'import_name':
    TypeError: Unsupported type for field 'import_name': str | None
ERROR tests/test_callbacks.py - huggingface_hub.errors.StrictDataclassFieldValidationError: Validation error for field 'import_name':
    TypeError: Unsupported type for field 'import_name': str | None
ERROR tests/test_callbacks.py - huggingface_hub.errors.StrictDataclassFieldValidationError: Validation error for field 'import_name':
    TypeError: Unsupported type for field 'import_name': str | None
ERROR tests/test_callbacks.py - huggingface_hub.errors.StrictDataclassFieldValidationError: Validation error for field 'import_name':
    TypeError: Unsupported type for field 'import_name': str | None
ERROR tests/test_callbacks.py - huggingface_hub.errors.StrictDataclassFieldValidationError: Validation error for field 'import_name':
    TypeError: Unsupported type for field 'import_name': str | None
ERROR tests/test_chat_template_utils.py - RuntimeError: Failed to import trl.chat_template_utils because of the following error (look up to see its traceback):
Validation error for field 'import_name':
    TypeError: Unsupported type for field 'import_name': str | None
ERROR tests/test_chat_template_utils.py - RuntimeError: Failed to import trl.chat_template_utils because of the following error (look up to see its traceback):
Validation error for field 'import_name':
    TypeError: Unsupported type for field 'import_name': str | None
ERROR tests/test_chat_template_utils.py - RuntimeError: Failed to import trl.chat_template_utils because of the following error (look up to see its traceback):
Validation error for field 'import_name':
    TypeError: Unsupported type for field 'import_name': str | None
ERROR tests/test_chat_template_utils.py - RuntimeError: Failed to import trl.chat_template_utils because of the following error (look up to see its traceback):
Validation error for field 'import_name':
    TypeError: Unsupported type for field 'import_name': str | None
ERROR tests/test_cli.py - huggingface_hub.errors.StrictDataclassFieldValidationError: Validation error for field 'import_name':
    TypeError: Unsupported type for field 'import_name': str | None
ERROR tests/test_cli.py - huggingface_hub.errors.StrictDataclassFieldValidationError: Validation error for field 'import_name':
    TypeError: Unsupported type for field 'import_name': str | None
ERROR tests/test_cli.py - huggingface_hub.errors.StrictDataclassFieldValidationError: Validation error for field 'import_name':
    TypeError: Unsupported type for field 'import_name': str | None
ERROR tests/test_cli.py - huggingface_hub.errors.StrictDataclassFieldValidationError: Validation error for field 'import_name':
    TypeError: Unsupported type for field 'import_name': str | None
ERROR tests/test_cli_utils.py - huggingface_hub.errors.StrictDataclassFieldValidationError: Validation error for field 'import_name':
    TypeError: Unsupported type for field 'import_name': str | None
ERROR tests/test_cli_utils.py - huggingface_hub.errors.StrictDataclassFieldValidationError: Validation error for field 'import_name':
    TypeError: Unsupported type for field 'import_name': str | None
ERROR tests/test_cli_utils.py - huggingface_hub.errors.StrictDataclassFieldValidationError: Validation error for field 'import_name':
    TypeError: Unsupported type for field 'import_name': str | None
ERROR tests/test_cli_utils.py - huggingface_hub.errors.StrictDataclassFieldValidationError: Validation error for field 'import_name':
    TypeError: Unsupported type for field 'import_name': str | None
ERROR tests/test_data_utils.py - huggingface_hub.errors.StrictDataclassFieldValidationError: Validation error for field 'import_name':
    TypeError: Unsupported type for field 'import_name': str | None
ERROR tests/test_data_utils.py - huggingface_hub.errors.StrictDataclassFieldValidationError: Validation error for field 'import_name':
    TypeError: Unsupported type for field 'import_name': str | None
ERROR tests/test_data_utils.py - huggingface_hub.errors.StrictDataclassFieldValidationError: Validation error for field 'import_name':
    TypeError: Unsupported type for field 'import_name': str | None
ERROR tests/test_data_utils.py - huggingface_hub.errors.StrictDataclassFieldValidationError: Validation error for field 'import_name':
    TypeError: Unsupported type for field 'import_name': str | None
ERROR tests/test_dpo_trainer.py - RuntimeError: Failed to import trl.trainer.dpo_trainer because of the following error (look up to see its traceback):
Validation error for field 'import_name':
    TypeError: Unsupported type for field 'import_name': str | None
ERROR tests/test_dpo_trainer.py - RuntimeError: Failed to import trl.trainer.dpo_trainer because of the following error (look up to see its traceback):
Validation error for field 'import_name':
    TypeError: Unsupported type for field 'import_name': str | None
ERROR tests/test_dpo_trainer.py - RuntimeError: Failed to import trl.trainer.dpo_trainer because of the following error (look up to see its traceback):
Validation error for field 'import_name':
    TypeError: Unsupported type for field 'import_name': str | None
ERROR tests/test_dpo_trainer.py - RuntimeError: Failed to import trl.trainer.dpo_trainer because of the following error (look up to see its traceback):
Validation error for field 'import_name':
    TypeError: Unsupported type for field 'import_name': str | None
ERROR tests/test_grpo_trainer.py - huggingface_hub.errors.StrictDataclassFieldValidationError: Validation error for field 'import_name':
    TypeError: Unsupported type for field 'import_name': str | None
ERROR tests/test_grpo_trainer.py - huggingface_hub.errors.StrictDataclassFieldValidationError: Validation error for field 'import_name':
    TypeError: Unsupported type for field 'import_name': str | None
ERROR tests/test_grpo_trainer.py - huggingface_hub.errors.StrictDataclassFieldValidationError: Validation error for field 'import_name':
    TypeError: Unsupported type for field 'import_name': str | None
ERROR tests/test_grpo_trainer.py - huggingface_hub.errors.StrictDataclassFieldValidationError: Validation error for field 'import_name':
    TypeError: Unsupported type for field 'import_name': str | None
ERROR tests/test_model_utils.py - huggingface_hub.errors.StrictDataclassFieldValidationError: Validation error for field 'import_name':
    TypeError: Unsupported type for field 'import_name': str | None
ERROR tests/test_model_utils.py - huggingface_hub.errors.StrictDataclassFieldValidationError: Validation error for field 'import_name':
    TypeError: Unsupported type for field 'import_name': str | None
ERROR tests/test_model_utils.py - huggingface_hub.errors.StrictDataclassFieldValidationError: Validation error for field 'import_name':
    TypeError: Unsupported type for field 'import_name': str | None
ERROR tests/test_model_utils.py - huggingface_hub.errors.StrictDataclassFieldValidationError: Validation error for field 'import_name':
    TypeError: Unsupported type for field 'import_name': str | None
ERROR tests/test_reward_trainer.py - RuntimeError: Failed to import trl.trainer.reward_trainer because of the following error (look up to see its traceback):
Validation error for field 'import_name':
    TypeError: Unsupported type for field 'import_name': str | None
ERROR tests/test_reward_trainer.py - RuntimeError: Failed to import trl.trainer.reward_trainer because of the following error (look up to see its traceback):
Validation error for field 'import_name':
    TypeError: Unsupported type for field 'import_name': str | None
ERROR tests/test_reward_trainer.py - RuntimeError: Failed to import trl.trainer.reward_trainer because of the following error (look up to see its traceback):
Validation error for field 'import_name':
    TypeError: Unsupported type for field 'import_name': str | None
ERROR tests/test_reward_trainer.py - RuntimeError: Failed to import trl.trainer.reward_trainer because of the following error (look up to see its traceback):
Validation error for field 'import_name':
    TypeError: Unsupported type for field 'import_name': str | None
ERROR tests/test_rewards.py - huggingface_hub.errors.StrictDataclassFieldValidationError: Validation error for field 'import_name':
    TypeError: Unsupported type for field 'import_name': str | None
ERROR tests/test_rewards.py - huggingface_hub.errors.StrictDataclassFieldValidationError: Validation error for field 'import_name':
    TypeError: Unsupported type for field 'import_name': str | None
ERROR tests/test_rewards.py - huggingface_hub.errors.StrictDataclassFieldValidationError: Validation error for field 'import_name':
    TypeError: Unsupported type for field 'import_name': str | None
ERROR tests/test_rewards.py - huggingface_hub.errors.StrictDataclassFieldValidationError: Validation error for field 'import_name':
    TypeError: Unsupported type for field 'import_name': str | None
ERROR tests/test_rich_progress_callback.py - huggingface_hub.errors.StrictDataclassFieldValidationError: Validation error for field 'import_name':
    TypeError: Unsupported type for field 'import_name': str | None
ERROR tests/test_rich_progress_callback.py - huggingface_hub.errors.StrictDataclassFieldValidationError: Validation error for field 'import_name':
    TypeError: Unsupported type for field 'import_name': str | None
ERROR tests/test_rich_progress_callback.py - huggingface_hub.errors.StrictDataclassFieldValidationError: Validation error for field 'import_name':
    TypeError: Unsupported type for field 'import_name': str | None
ERROR tests/test_rich_progress_callback.py - huggingface_hub.errors.StrictDataclassFieldValidationError: Validation error for field 'import_name':
    TypeError: Unsupported type for field 'import_name': str | None
ERROR tests/test_rloo_trainer.py - RuntimeError: Failed to import trl.trainer.rloo_trainer because of the following error (look up to see its traceback):
Validation error for field 'import_name':
    TypeError: Unsupported type for field 'import_name': str | None
ERROR tests/test_rloo_trainer.py - RuntimeError: Failed to import trl.trainer.rloo_trainer because of the following error (look up to see its traceback):
Validation error for field 'import_name':
    TypeError: Unsupported type for field 'import_name': str | None
ERROR tests/test_rloo_trainer.py - RuntimeError: Failed to import trl.trainer.rloo_trainer because of the following error (look up to see its traceback):
Validation error for field 'import_name':
    TypeError: Unsupported type for field 'import_name': str | None
ERROR tests/test_rloo_trainer.py - RuntimeError: Failed to import trl.trainer.rloo_trainer because of the following error (look up to see its traceback):
Validation error for field 'import_name':
    TypeError: Unsupported type for field 'import_name': str | None
ERROR tests/test_sft_trainer.py - huggingface_hub.errors.StrictDataclassFieldValidationError: Validation error for field 'import_name':
    TypeError: Unsupported type for field 'import_name': str | None
ERROR tests/test_sft_trainer.py - huggingface_hub.errors.StrictDataclassFieldValidationError: Validation error for field 'import_name':
    TypeError: Unsupported type for field 'import_name': str | None
ERROR tests/test_sft_trainer.py - huggingface_hub.errors.StrictDataclassFieldValidationError: Validation error for field 'import_name':
    TypeError: Unsupported type for field 'import_name': str | None
ERROR tests/test_sft_trainer.py - huggingface_hub.errors.StrictDataclassFieldValidationError: Validation error for field 'import_name':
    TypeError: Unsupported type for field 'import_name': str | None
ERROR tests/test_utils.py - huggingface_hub.errors.StrictDataclassFieldValidationError: Validation error for field 'import_name':
    TypeError: Unsupported type for field 'import_name': str | None
ERROR tests/test_utils.py - huggingface_hub.errors.StrictDataclassFieldValidationError: Validation error for field 'import_name':
    TypeError: Unsupported type for field 'import_name': str | None
ERROR tests/test_utils.py - huggingface_hub.errors.StrictDataclassFieldValidationError: Validation error for field 'import_name':
    TypeError: Unsupported type for field 'import_name': str | None
ERROR tests/test_utils.py - huggingface_hub.errors.StrictDataclassFieldValidationError: Validation error for field 'import_name':
    TypeError: Unsupported type for field 'import_name': str | None
ERROR tests/test_vllm_client_server.py - huggingface_hub.errors.StrictDataclassFieldValidationError: Validation error for field 'import_name':
    TypeError: Unsupported type for field 'import_name': str | None
ERROR tests/test_vllm_client_server.py - huggingface_hub.errors.StrictDataclassFieldValidationError: Validation error for field 'import_name':
    TypeError: Unsupported type for field 'import_name': str | None
ERROR tests/test_vllm_client_server.py - huggingface_hub.errors.StrictDataclassFieldValidationError: Validation error for field 'import_name':
    TypeError: Unsupported type for field 'import_name': str | None
ERROR tests/test_vllm_client_server.py - huggingface_hub.errors.StrictDataclassFieldValidationError: Validation error for field 'import_name':
    TypeError: Unsupported type for field 'import_name': str | None
======================== 73 passed, 68 errors
```

Stacktrace:
```python
 ______________ ERROR collecting tests/test_vllm_client_server.py _______________
.venv/lib/python3.12/site-packages/huggingface_hub/dataclasses.py:131: in __strict_setattr__
    validator(value)
.venv/lib/python3.12/site-packages/huggingface_hub/dataclasses.py:435: in validator
    type_validator(field.name, value, field.type)
.venv/lib/python3.12/site-packages/huggingface_hub/dataclasses.py:332: in type_validator
    raise TypeError(f"Unsupported type for field '{name}': {expected_type}")
E   TypeError: Unsupported type for field 'import_name': str | None

The above exception was the direct cause of the following exception:
tests/test_vllm_client_server.py:21: in <module>
    from transformers import AutoModelForCausalLM, AutoProcessor, AutoTokenizer
<frozen importlib._bootstrap>:1412: in _handle_fromlist
    ???
.venv/lib/python3.12/site-packages/transformers/utils/import_utils.py:2302: in __getattr__
    module = self._get_module(self._class_to_module[name])
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
.venv/lib/python3.12/site-packages/transformers/utils/import_utils.py:2332: in _get_module
    raise e
.venv/lib/python3.12/site-packages/transformers/utils/import_utils.py:2330: in _get_module
    return importlib.import_module("." + module_name, self.__name__)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/__w/_tool/Python/3.12.13/x64/lib/python3.12/importlib/__init__.py:90: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
.venv/lib/python3.12/site-packages/transformers/models/auto/processing_auto.py:28: in <module>
    from ...processing_utils import ProcessorMixin
.venv/lib/python3.12/site-packages/transformers/processing_utils.py:75: in <module>
    from .modeling_utils import PreTrainedAudioTokenizerBase
.venv/lib/python3.12/site-packages/transformers/modeling_utils.py:58: in <module>
    from .integrations.hub_kernels import is_kernel, load_and_register_kernel
.venv/lib/python3.12/site-packages/transformers/integrations/hub_kernels.py:23: in <module>
    from kernels import (
.venv/lib/python3.12/site-packages/kernels/__init__.py:7: in <module>
    from kernels.layer import (
.venv/lib/python3.12/site-packages/kernels/layer/__init__.py:2: in <module>
    from .func import (
.venv/lib/python3.12/site-packages/kernels/layer/func.py:11: in <module>
    from ..utils import (
.venv/lib/python3.12/site-packages/kernels/utils.py:22: in <module>
    from kernels.deps import validate_dependencies
.venv/lib/python3.12/site-packages/kernels/deps.py:62: in <module>
    _DEPENDENCY_DATA = DependencyData.from_dict(json.load(f))
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
.venv/lib/python3.12/site-packages/kernels/deps.py:48: in from_dict
    name: DependencyInfo.from_dict(info)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
.venv/lib/python3.12/site-packages/kernels/deps.py:35: in from_dict
    python=[PythonPackage.from_dict(p) for p in data.get("python", [])],
            ^^^^^^^^^^^^^^^^^^^^^^^^^^
.venv/lib/python3.12/site-packages/kernels/deps.py:19: in from_dict
    return PythonPackage(
.venv/lib/python3.12/site-packages/huggingface_hub/dataclasses.py:230: in init_with_validate
    initial_init(self, *args, **kwargs)  # type: ignore [call-arg]
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
<string>:4: in __init__
    ???
.venv/lib/python3.12/site-packages/huggingface_hub/dataclasses.py:133: in __strict_setattr__
    raise StrictDataclassFieldValidationError(field=name, cause=e) from e
E   huggingface_hub.errors.StrictDataclassFieldValidationError: Validation error for field 'import_name':
E       TypeError: Unsupported type for field 'import_name': str | None
```

--- a/.github/workflows/tests.yml
+++ b/.github/workflows/tests.yml
@@ -132,11 +132,12 @@ jobs:
       - name: Install dependencies
         run: |
           source .venv/bin/activate
-          uv pip install ".[dev]"
-          uv pip install -U git+https://github.com/huggingface/accelerate.git
-          uv pip install -U git+https://github.com/huggingface/datasets.git
-          uv pip install -U git+https://github.com/huggingface/transformers.git
-          uv pip install -U git+https://github.com/huggingface/peft.git
+          uv pip install -U \
+            ".[dev]" \
+            "accelerate @ git+https://github.com/huggingface/accelerate.git" \
+            "datasets @ git+https://github.com/huggingface/datasets.git" \
+            "transformers @ git+https://github.com/huggingface/transformers.git" \
+            "peft @ git+https://github.com/huggingface/peft.git"
 
       - name: Test with pytest
         run: |
@@ -240,10 +241,11 @@ jobs:
       - name: Install dependencies
         run: |
           source .venv/bin/activate
-          uv pip install ".[dev]"
-          uv pip install accelerate==1.4.0
-          uv pip install datasets==3.0.0
-          uv pip install transformers==4.56.2
+          uv pip install \
+            ".[dev]" \
+            "accelerate==1.4.0" \
+            "datasets==3.0.0" \
+            "transformers==4.56.2"
 
       - name: Test with pytest
         run: |

--- a/.github/workflows/tests.yml
+++ b/.github/workflows/tests.yml
@@ -244,7 +244,7 @@ jobs:
           uv pip install \
             ".[dev]" \
             "accelerate==1.4.0" \
-            "datasets==3.0.0" \
+            "datasets==4.7.0" \
             "transformers==4.56.2"
 
       - name: Test with pytest

--- a/.github/workflows/tests_latest.yml
+++ b/.github/workflows/tests_latest.yml
@@ -49,10 +49,11 @@ jobs:
       - name: Install dependencies
         run: |
           source .venv/bin/activate
-          uv pip install ".[dev]"
-          uv pip install -U git+https://github.com/huggingface/accelerate.git
-          uv pip install -U git+https://github.com/huggingface/datasets.git
-          uv pip install -U git+https://github.com/huggingface/transformers.git
+          uv pip install -U \
+            ".[dev]" \
+            "accelerate @ git+https://github.com/huggingface/accelerate.git" \
+            "datasets @ git+https://github.com/huggingface/datasets.git" \
+            "transformers @ git+https://github.com/huggingface/transformers.git"
 
       - name: Test with pytest
         run: |

--- a/.github/workflows/tests_transformers_branch.yml
+++ b/.github/workflows/tests_transformers_branch.yml
@@ -51,8 +51,9 @@ jobs:
       - name: Install dependencies
         run: |
           source .venv/bin/activate
-          uv pip install ".[dev]"
-          uv pip install -U git+https://github.com/huggingface/transformers.git@${{ inputs.transformers_ref }}
+          uv pip install -U \
+            ".[dev]" \
+            "transformers @ git+https://github.com/huggingface/transformers.git@${{ inputs.transformers_ref }}"
 
       - name: Test with pytest
         run: |
@@ -105,8 +106,9 @@ jobs:
       - name: Install dependencies
         run: |
           source .venv/bin/activate
-          uv pip install ".[dev]"
-          uv pip install -U git+https://github.com/huggingface/transformers.git@${{ inputs.transformers_ref }}
+          uv pip install -U \
+            ".[dev]" \
+            "transformers @ git+https://github.com/huggingface/transformers.git@${{ inputs.transformers_ref }}"
 
       - name: Run distributed smoke tests
         run: |

